### PR TITLE
[BinarySerialization] Refactor the existing logic

### PR DESCRIPTION
**Pull request by Buga1chukY**

### CODE REVIEW

File: src/main/java/com/knu/buga1chuk/serialization/ReadObject.java [hunk 1]
```diff
@@ -2,31 +2,34 @@
...
 import com.knu.buga1chuk.constant.FilePathConstants;
 import com.knu.buga1chuk.model.Person;
+import com.knu.buga1chuk.service.PersonBinaryService;
+import com.knu.buga1chuk.service.PersonService;
 
-import java.io.*;
+import java.io.File;
+import java.util.List;
 
 public class ReadObject {
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/serialization/ReadObject.java:11]: Change title to `ReadPersonBinaryDemo`

File: src/main/java/com/knu/buga1chuk/serialization/WriteObject.java [hunk 2]
```diff
@@ -2,28 +2,21 @@
...
+import com.knu.buga1chuk.service.PersonBinaryService;
+import com.knu.buga1chuk.service.PersonService;
 
 import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.ObjectOutputStream;
+import java.util.List;
 
 public class WriteObject {
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/serialization/WriteObject.java:11]: Change title to `WritePersonBinaryDemo`

File: src/main/java/com/knu/buga1chuk/serialization/WriteObject.java
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/serialization/WriteObject.java:1]: Change package title to binary

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            oos.writeObject(personList);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static List<Person> getPersonFromFile(File file) {
+        List<Person> persons = new ArrayList<>();
+        PersonList persons1 = new PersonList(persons);
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:27]: Rename to `personList`

Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:27]: Remove the assignment to the personList.

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import com.knu.buga1chuk.model.Person;
+import com.knu.buga1chuk.model.PersonList;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PersonBinaryService {
+
+    public static void writePerson(File file, List<Person> persons) {
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:12]: Remove static

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+             ObjectOutputStream oos = new ObjectOutputStream(fos)) {
+
+            oos.writeObject(personList);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static List<Person> getPersonFromFile(File file) {
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:25]: Remove static

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/com/knu/buga1chuk/serialization/ReadObject.java (hunk 1 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/serialization/ReadObject.java:11]: fix

File: src/main/java/com/knu/buga1chuk/serialization/WriteObject.java (hunk 2 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/serialization/WriteObject.java:11]: fix

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java (hunk 3 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:27]: fix

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java (hunk 4 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:12]: fix

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java (hunk 5 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:25]: fix

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        PersonList personList;
+
+        try (FileInputStream fis = new FileInputStream(file);
+             ObjectInputStream ois = new ObjectInputStream(fis)) {
+
+            personList = (PersonList) ois.readObject();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new IllegalStateException();
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:34]: You lost the initial exception.

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        } catch (ClassNotFoundException e) {
+            System.out.println("There is no such class");
+            e.printStackTrace();
+            throw new IllegalStateException();
```
Comment (MLGMAG) [inline src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:38]: You lost the initial exception.

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/com/knu/buga1chuk/serialization/WriteObject.java
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/serialization/WriteObject.java:1]: fix

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java (hunk 6 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:34]: fix

File: src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java (hunk 7 above)
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/service/PersonBinaryService.java:38]: fix

File: src/main/java/com/knu/buga1chuk/serialization/WriteObject.java
Comment (Buga1chukY) [reply src/main/java/com/knu/buga1chuk/serialization/WriteObject.java:1]: fix

**Review by MLGMAG — APPROVED** [review]

**Review by MLGMAG — CHANGES_REQUESTED** [review]
Squash and I will approve

**Review by MLGMAG — APPROVED** [review]

**Pull request merged**